File: leetcode/src/main/ruby/container_with_most_water.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
    area, i, j = 0, 0, height.length - 1
    while i < j
        area = [area, (j - i) * [height[i], height[j]].min].max
        if height[i] < height[j]
            i += 1
        else
            j -= 1
        end
    end
    area
end

File: leetcode/src/main/golang/container_with_most_water.go
func maxArea(height []int) int {
    max := 0
    for i, j := 0, len(height) - 1; i < j; {
        max = iMax(max, (j - i) * iMin(height[i], height[j]));
        if height[i] < height[j] {
            i++
        } else {
            j--
        }
    }
    return max
}

func iMax(a, b int) int {
    if a > b {
        return a
    } else {
        return b
    }
}

func iMin(a, b int) int {
    if a > b {
        return b
    } else {
        return a
    }
}